cmd/aws/create.go

```diff
@@ -161,7 +156,125 @@ func ValidateProvidedFlags(gitProvider string) error {
 		log.Info().Msgf("%q %s", "gitlab.com", key.Type())
 	}
 
+	if err := ValidateAMIType(cfg, amiType, nodeType); err != nil {
+		progress.Error(err.Error())
+		return fmt.Errorf("failed to validte ami type for node group: %w", err)
+	}
+
 	progress.CompleteStep("Validate provided flags")
 
 	return nil
 }
+
+func ValidateAWSRegionAndRetrieveCredentials(cfg aws.Config) (*aws.Credentials, error) {
+	// Validate region by creating a client
+	stsClient := sts.NewFromConfig(cfg)
+	_, err := stsClient.GetCallerIdentity(context.TODO(), &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to validate AWS region: %w", err)
+	}
+
+	// Retrieve credentials
+	creds, err := cfg.Credentials.Retrieve(context.TODO())
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve AWS credentials: %w", err)
+	}
+
+	return &creds, nil
+}
+
+func ValidateAMIType(cfg aws.Config, amiType, nodeType string) error {
+	ssmTypes := map[string]string{
+		"AL2_x86_64":                 "/aws/service/eks/optimized-ami/1.29/amazon-linux-2/recommended/image_id",
+		"AL2_ARM_64":                 "/aws/service/eks/optimized-ami/1.29/amazon-linux-2-arm64/recommended/image_id",
+		"BOTTLEROCKET_ARM_64":        "/aws/service/bottlerocket/aws-k8s-1.29/arm64/latest/image_id",
+		"BOTTLEROCKET_x86_64":        "/aws/service/bottlerocket/aws-k8s-1.29/x86_64/latest/image_id",
+		"BOTTLEROCKET_ARM_64_NVIDIA": "/aws/service/bottlerocket/aws-k8s-1.29-nvidia/arm64/latest/image_id",
+		"BOTTLEROCKET_x86_64_NVIDIA": "/aws/service/bottlerocket/aws-k8s-1.29-nvidia/x86_64/latest/image_id",
+	}
+
+	_, ok := ssmTypes[amiType]
+	if !ok {
+		return fmt.Errorf("not a valid ami type")
+	}
+
+	ssmClient := ssm.NewFromConfig(cfg)
+	ssmParameterName := ssmTypes[amiType]
+
+	amiID, err := GetLatestAMIFromSSM(ssmClient, ssmParameterName)
+	if err != nil {
+		return fmt.Errorf("failed to get AMI ID from SSM: %w", err)
+	}
+	log.Info().Msgf("ami type is  %s", amiType)
+
+	architecture, err := GetAMIArchitecture(cfg, amiID)
+	if err != nil {
+		return fmt.Errorf("failed to get AMI architecture: %w", err)
+	}
+
+	instanceTypes, err := GetSupportedInstanceTypes(cfg, architecture)
+	if err != nil {
+		return fmt.Errorf("failed to get supported instance types: %w", err)
+	}
+
+	for _, instanceType := range instanceTypes {
+		if instanceType == nodeType {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("node type %s not supported for %s", nodeType, amiType)
+}
+
+func GetLatestAMIFromSSM(ssmClient *ssm.Client, parameterName string) (string, error) {
+	input := &ssm.GetParameterInput{
+		Name: aws.String(parameterName),
+	}
+	output, err := ssmClient.GetParameter(context.TODO(), input)
+	if err != nil {
+		return "", fmt.Errorf("failed to initialise ssm client: %w", err)
+	}
+
+	return *output.Parameter.Value, nil
+}
+
+func GetAMIArchitecture(cfg aws.Config, amiID string) (string, error) {
+	ec2Client := ec2.NewFromConfig(cfg)
```

Comment:
Similar to other clients, this should be passed in as a dependency and not instantiated in-line. Makes it basically impossible to test or for anyone else to mock it if necessary.